### PR TITLE
2.6 review, fix memleaks/timeouts, improve message processing (non blocking recv)

### PR DESCRIPTION
**Pull request by FrancoisRegisDegott-eaton**
* sensor_list.cc: fix memleaks, fix send timeout
* messaging: use a poller/timeout to recv request response (non blocking)...
* actors: rework/simplify messages processing
* ASSET_DETAILS recv, fix possible infinite loop (while(!uuids.empty()))  (fty_nut_server.cc)
* review

FrancoisRegisDegott@eaton.com

### CODE REVIEW

File: lib/src/alert_actor.cc [hunk 1]
```diff
@@ -26,58 +26,45 @@
...
+            char* end;
+            timeout_ms = std::strtoul(polling, &end, 10) * 1000;
+            if (timeout_ms == 0) {
+                log_error("aa: invalid POLLING value '%s', using default instead", polling);
+                timeout_ms = 30000;
+            }
         }
+        log_debug("aa: timeout: %zu ms", timeout_ms);
         zstr_free(&polling);
-    } else if (streq(cmd, ACTION_CONFIGURE)) {
```
Comment (ericclappier-eaton) [inline lib/src/alert_actor.cc:65]: May be used in the future ...

File: lib/src/fty_nut_server.cc [hunk 2]
```diff
@@ -68,103 +88,174 @@ static bool get_initial_licensing(StateManager::Writer& state_writer, mlm_client
...
+        }
     }
-    if (changed)
-        state_writer.commit();
-    log_info("Initial ASSETS request complete (%zd/%zd powerdevices, %zd/%zd sensors)",
-        state_writer.getState().getPowerDevices().size(), state_writer.getState().getAllPowerDevices().size(),
-        state_writer.getState().getSensors().size(), state_writer.getState().getAllSensors().size());
-}
-
-uint64_t polling_timeout(uint64_t last_poll, uint64_t polling_timeout)
```
Comment (ericclappier-eaton) [inline lib/src/fty_nut_server.cc:154]: Protection for many devices not needed ?

**Review by ericclappier-eaton — APPROVED** [review]

File: lib/src/fty_nut_server.cc (hunk 2 above)
Comment (FrancoisRegisDegott-eaton) [reply lib/src/fty_nut_server.cc:154]: Yes, i changed that... the previous effect was to focus on polling action, loosing reactivity on notification messages processing.
Anyway, the case is rarely reached, and don't changed poll/message processing even n our endurance setup...

File: lib/src/alert_actor.cc (hunk 1 above)
Comment (FrancoisRegisDegott-eaton) [reply lib/src/alert_actor.cc:65]: a few obvious lines to add in that case.  ;)

**Review by perrettecl — APPROVED** [review]

**Pull request merged**